src/maison/config.py

```diff
@@ -189,4 +189,31 @@ def _generate_config_dict(self) -> Dict[Any, Any]:
             return self._sources[0].to_dict()
 
         source_dicts = [source.to_dict() for source in self._sources]
-        return reduce(lambda a, b: {**a, **b}, source_dicts)
+        return reduce(lambda a, b: deep_merge(a, b), source_dicts)
+
+
+def deep_merge(destination, source):
```

Comment:
missing a little bit of formatting here:
- arg types
- return type
- doc string style (see the [google style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods))

Reply:
also, i think this function should live in the `utils` module

Reply:
Done in https://github.com/dbatten5/maison/pull/89/commits/6cac2a4092440a7f741ad53d7ca7601e94a3ad07